connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java

```diff
@@ -118,4 +81,49 @@ public static void main(String[] args) throws Exception {
             Exit.exit(2);
         }
     }
+
+    public Connect startConnect(Map<String, String> workerProps) {
+        log.info("Scanning for plugin classes. This might take a moment ...");
+        Plugins plugins = new Plugins(workerProps);
+        plugins.compareAndSwapWithDelegatingLoader();
+        DistributedConfig config = new DistributedConfig(workerProps);
```

Comment:
It might be even more valuable to break this method into multiple protected methods and to have more of the components created in this method be members. However, that's probably work for another ticket.